File: ex4/task4.2/MaxTwoDoubles.java
/**
* Prints the max between two doubles
* Tested for (7.4;7.3), (2;6.3), (3.2;3.2).
* Exception given for mini 12 
* @author Vlad Georgescu 
*/
public class MaxTwoDoubles
{
    public static void main(String[] args)
    {
        double firstDouble = Double.parseDouble(args[0]);
        double secondDouble = Double.parseDouble(args[1]);
         
        System.out.println("max(" + firstDouble + ", " + secondDouble + "):"); 
        
        if (firstDouble > secondDouble) {
            System.out.println(firstDouble);
        } else {
            System.out.println(secondDouble);
        }
    }
}

File: Ex4/Task4.2/MaxTwoDoubles.scala
/**
 * Takes two command line arguments and prints the
 * larger number of the two
 */
object MaxTwoDoubles {
    def main(args: Array[String]) {
        val n1: Double = args(0).toInt
        val n2: Double = args(1).toInt

        val max: Double = if (n1 > n2) n1 else n2

        println(s"Numbers $n1 and $n2 -- max is $max")
    }
}